ocaml/typing/typedecl.ml

```diff
@@ -1548,8 +1548,6 @@ let transl_type_decl env rec_flag sdecl_list =
         raise (Error (loc, Type_clash (new_env, err))))
       checks)
     delayed_jkind_checks;
-  (* Check that constraints are enforced *)
-  List.iter2 (check_constraints new_env) sdecl_list decls;
   (* Check that all type variables are closed; this also defaults any remaining
```

Comment:
Should this comment be updated to include something like `Defaulting also has to occur prior to check_constraints to prevent soundness bugs`?

Reply:
Good point. Done.